src/rebase.rs

```diff
@@ -0,0 +1,487 @@
+    type Raw = *mut raw::git_rebase_operation;
+    unsafe fn from_raw(raw: *mut raw::git_rebase_operation) -> RebaseOperation<'rebase, 'repo> {
+        RebaseOperation {
+            raw: raw,
+            _marker: marker::PhantomData,
+        }
+    }
+    fn raw(&self) -> *mut raw::git_rebase_operation {
+        self.raw
+    }
+}
+
+impl<'rebase, 'repo: 'rebase> Debug for RebaseOperation<'rebase, 'repo> {
+    fn fmt(&self, f: &mut Formatter) -> FmtResult {
+        f.debug_struct("RebaseOperation")
+            .field("id", &self.id())
+            .field("kind", &self.kind())
+            .finish()
+    }
+}
+
+impl<'a> RebaseOptions<'a> {
+    /// Initiate default set of rebase options.
+    pub fn new() -> RebaseOptions<'a> {
+        RebaseOptions {
+            version: raw::GIT_REBASE_OPTIONS_VERSION as usize,
+            quiet: false,
+            in_memory: false,
+            rewrite_notes_ref: None,
+            merge_options: None,
+            checkout_builder: None,
+        }
+    }
+
+    /// this will instruct other clients working
+    /// on this `rebase` that you want a quiet rebase experience.
+    /// This is provided for interoperability between Git tools
+    pub fn quiet(&mut self, quiet: bool) -> &mut Self {
+        self.quiet = quiet;
+        self
+    }
+
+    /// Perform an in-memory rebase, will not updated the repository to be in a rebasing-state
+    /// or modify the working directory.
+    pub fn in_memory(&mut self, in_memory: bool) -> &mut Self {
+        self.in_memory = in_memory;
+        self
+    }
+
+    /// The name of the notes reference used to rewrite notes
+    /// for rebased commits when finishing the rebase
+    pub fn rewrite_notes_ref(&mut self, rewrite_notes_ref: Option<CString>) -> &mut Self {
+        self.rewrite_notes_ref = rewrite_notes_ref;
+        self
+    }
+
+    /// Options to control how trees are merged during a `rebase`.
+    pub fn merge_options(&mut self, merge_options: Option<MergeOptions>) -> &mut Self {
+        self.merge_options = merge_options;
+        self
+    }
+
+    /// Options to control how files are written during a `rebase`.
+    pub fn checkout_builder(&mut self, checkout_builder: Option<CheckoutBuilder<'a>>) -> &mut Self {
+        self.checkout_builder = checkout_builder;
+        self
+    }
+
+    /// raw value of options
+    pub unsafe fn raw(&mut self) -> raw::git_rebase_options {
+        let mut checkout_options: raw::git_checkout_options = mem::zeroed();
+        raw::git_checkout_init_options(&mut checkout_options, raw::GIT_CHECKOUT_OPTIONS_VERSION);
+        if let Some(ref mut checkout_builder) = self.checkout_builder {
+            checkout_builder.configure(&mut checkout_options);
+        }
+        let mut merge_options: raw::git_merge_options = mem::zeroed();
+        if let Some(ref opts) = self.merge_options {
+            ptr::copy(opts.raw(), &mut merge_options, 1);
```

Comment:
I think this can be `merge_options = opts.raw()` perhaps?

Reply:
I had to deref it from a `*const` but `merge_options = *opts.raw();` seems semantically correct?

Reply:
Oh wait no I get `cannot move out of dereference of raw pointer` on compile (but not from rls). The `git_merge_options` struct seems like it would be safe to derive copy/clone on, and there are a few Copy types in `libgit2-sys/lib.rs`, so I've done that. Happy to change it back, though.